addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -849,63 +745,117 @@ options.registry.facebookPage = options.Class.extend({
                 limit: 1,
             }).then(function (res) {
                 if (res) {
-                    self.fbData.href = res[0].social_facebook || 'https://www.facebook.com/Odoo';
+                    self.fbData.href = res[0].social_facebook || '';
                 }
             }));
         }
 
-        return Promise.all(defs);
-    },
-    /**
-     * @override
-     */
-    start: function () {
-        var self = this;
-        this.$target.on('click.facebook_page_option', '.o_add_facebook_page', function (ev) {
-            ev.preventDefault();
-            self.fbPageOptions();
-        });
-        return this._super.apply(this, arguments);
-    },
-    /**
-     * @override
-     */
-    destroy: function () {
-        this._super.apply(this, arguments);
-        this.$target.off('.facebook_page_option');
+        return Promise.all(defs).then(() => this._markFbElement());
     },
 
     //--------------------------------------------------------------------------
     // Options
     //--------------------------------------------------------------------------
 
     /**
-     * Opens a dialog to configure the facebook page options.
+     * Toggles a tab.
+     *
+     * @see this.selectClass for parameters
+     * @param {String} optionName the name of the option to toggle
+     */
+    toggleOption: function (previewMode, widgetValue, {optionName}) {
+        if (optionName.startsWith('tab.')) {
+            optionName = optionName.replace('tab.', '');
+            if (widgetValue) {
+                this.fbData.tabs = this.fbData.tabs
+                    .split(',')
+                    .filter(t => t !== '')
+                    .concat([optionName])
+                    .join(',');
+            } else {
+                this.fbData.tabs = this.fbData.tabs
+                    .split(',')
+                    .filter(t => t !== optionName)
+                    .join(',');
+            }
+        } else {
+            this.fbData[optionName] = widgetValue;
+        }
+        this._markFbElement();
```

Comment:
Mmh... I guess I will have to make option methods async. With this branch is it theoretically possible to break the system.

Reply:
should I return the promise of _markFbElement then?

Reply:
Yes as soon as the branch I mentioned in discuss is merged, this will kinda be important

Reply:
I made that change preemptively so it should be fine already. Should be ready to merge then.